backend/src/main/java/fk/prof/backend/http/LeaderHttpVerticle.java

```diff
@@ -226,7 +226,8 @@ private void handleGetPolicy(RoutingContext context) {
       Recorder.ProcessGroup pg = parseProcessGroup(context);
       PolicyDTO.VersionedPolicyDetails versionedPolicyDetails = policyStore.getVersionedPolicy(pg);
       if (versionedPolicyDetails == null) {
-        context.response().setStatusCode(404).end("Policy not found for process group " + RecorderProtoUtil.processGroupCompactRepr(pg));
+        HttpFailure httpFailure = new HttpFailure("Policy not found for process group " + RecorderProtoUtil.processGroupCompactRepr(pg), 404);
```

Comment:
just do a throw new HttpFailure here. it will be caught in catch below

Reply:
Made it throw the exception.